docs/fundamentals/networking/telemetry/metrics.md

```diff
@@ -0,0 +1,193 @@
+
+:::code language="yaml" source="snippets/metrics/prometheus.yml" highlight="31-99":::
+
+#### Start prometheus
+
+1. Reload the configuration or restart the Prometheus server.
+1. Confirm that OpenTelemetryTest is in the UP state in the **Status** > **Targets** page of the Prometheus web portal.
+![Prometheus status](~/docs/core/diagnostics/media/prometheus-status.png)
+
+1. On the Graph page of the Prometheus web portal, enter `http` in the expression text box and select `http_client_active_requests`
+![http_client_active_requests](~/docs/fundamentals/networking/telemetry/media/prometheus-search.png)
+  In the graph tab, Prometheus shows the value of the `http.client.active_requests` Counter that is being emitted by the example app.
+  ![Prometheus active requests graph](~/docs/fundamentals/networking/telemetry/media/prometheus-active-requests.png)
+
+#### Show metrics on a Grafana dashboard
+
+1. Follow the [standard instructions](https://prometheus.io/docs/visualization/grafana/#installing) to install Grafana and connect it to a Prometheus data source.
+
+1. Create a Grafana dashboard by clicking the **+** icon on the top toolbar then select **Dashboard**. In the dashboard editor that appears, enter **Open HTTP/1.1 Connections** in **Title** input box and the following query in the PromQL expression field:
+
+```
+sum by(http_connection_state) (http_client_open_connections{network_protocol_version="1.1"})
+```
+
+![Grafana HTTP/1.1 Connections](~/docs/fundamentals/networking/telemetry/media/grafana-connections.png)
+
+1. Click **Apply** to save and view the new dashboard. It displays the number of active vs idle HTTP/1.1 connections in the pool.
+
+## Enrichment
+
+Enrichment means the addition of custom tags (attributes) to a metric. This is useful if an app wants to add a custom categorization to dashboards or alerts built with metrics.
+The [`http.client.request.duration`](../../../core/diagnostics/built-in-metrics-system-net.md#instrument-httpclientrequestduration) instrument supports enrichment by registering callbacks with the <xref:System.Net.Http.Metrics.HttpMetricsEnrichmentContext>.
+Note that this is a low-level API and a separate callback registration is needed for each `HttpRequestMessage`.
+
+A simple way to do the callback registration at a single place is to implement a custom delegating <xref:System.Net.Http.DelegatingHandler>.
+This will allow you to intercept and modify the requests before they are forwarded to the inner handler and sent to the server:
+
+:::code language="csharp" source="snippets/metrics/Program.cs" id="snippet_Enrichment":::
+
+In case you are working with [`IHttpClientFactory`](../../../core/extensions/httpclient-factory.md), you can use xref:Microsoft.Extensions.DependencyInjection.HttpClientBuilderExtensions.AddHttpMessageHandler%2A to register the `EnrichmentHandler`:
```

Comment:
```suggestion
In case you are working with [`IHttpClientFactory`](../../../core/extensions/httpclient-factory.md), you can use <xref:Microsoft.Extensions.DependencyInjection.HttpClientBuilderExtensions.AddHttpMessageHandler%2A> to register the `EnrichmentHandler`:
```